yt_dlp/extractor/plex.py

```diff
@@ -0,0 +1,451 @@
+            'view_count': int,
+            'title': 'If you\'re happy and you know it',
+            'episode_number': 1,
+            'season_number': 1,
+            'episode': 'Episode 1',
+            'display_id': 'funniest-pets-and-people',
+            'thumbnail': 'https://cf-images.us-east-1.prod.boltdns.net/v1/jit/6058083015001/d958c52a-3e73-4902-8623-adfe2f36ea3f/main/1280x720/11m15s61ms/match/image.jpg',
+            'season': 'Season 1',
+        },
+        'params': {
+            'skip_download': True,
+        }
+    }, {
+        # season
+        'url': 'https://app.plex.tv/desktop/#!/provider/tv.plex.provider.vod/details?key=%2Flibrary%2Fmetadata%2F62a8b77b93fc109a6d020761&context=library%3Ahub.movies.reality-tv~8~9',
+        'info_dict': {
+            'id': '62a8b77b93fc109a6d020761',
+            'title': 'Funniest Pets & People',
+            'season': 'Funniest Pets & People',
+            'season_number': '1',
+            'thumbnail': 'https://image.tmdb.org/t/p/original/ngm14GVJ6jULL3zKK6puuVagRLH.jpg',
+        },
+        'playlist_count': 15,
+    }, {
+        # Extras
+        'url': 'https://app.plex.tv/desktop/#!/provider/tv.plex.provider.metadata/details?key=%2Flibrary%2Fmetadata%2F5ef5ee0d1ce3fd004039976a&context=library%3Ahub.home.top_watchlisted~4~1',
+        'info_dict': {
+            'id': '5ef5ee0d1ce3fd004039976a',
+            'title': 'Lightyear',
+            'cast': 'count:34',
+            'thumbnail': r're:https://image\.tmdb\.org/t/p/original/\w+\.jpg',
+            'duration': 6000,
+            'rating': 10,
+            'average_rating': 7.5,
+        },
+        'playlist_count': 22,
+    }]
+
+    def _real_extract(self, url):
+        provider, key, display_id = self._match_valid_url(url).group('provider', 'key', 'id')
+        key = urllib.parse.unquote(key)
+        media_json = self._download_json(
+            f'{self._CDN_ENDPOINT[provider]}{key}', display_id, query={'uri': f'provider://{provider}{key}', 'X-Plex-Token': self._TOKEN},
+            headers={'Accept': 'application/json'})['MediaContainer']['Metadata'][0]
+
+        # check if publicPagesURL, if exists redirect to PlexWatch*IE, else handle manually
+        if media_json.get('publicPagesURL'):
+            self.write_debug('got publicPagesURL, redirect to PlexWatch*IE')
+
+            additional_info = {
+                'view_count': int_or_none(media_json.get('viewCount')),
+                'thumbnail': media_json.get('thumb'),
+                'duration': int_or_none(media_json.get('duration'), 1000),
+                'cast': traverse_obj(media_json, ('Role', ..., 'tag')),
+                'rating': parse_age_limit(media_json.get('contentRating')),
+                'average_rating': float_or_none(media_json.get('rating')),
+            }
+            return self.url_result(media_json.get('publicPagesURL'), url_transparent=True, **additional_info)
```

Comment:
These metadata seems not redirected to the ie. In this commit, `test_PlexApp` and `test_PlexApp_1` still fail. Issue still exists even with `url_transparent=False`

Reply:
Not fixed yet, but test in PlexWatchMovie and PlexWatchEpisode should be fine now as the PlexBase Extractor set `media_json` as primary metadata source, implemented in https://github.com/yt-dlp/yt-dlp/pull/4366/commits/2a59212cc5ab176d81b17af20b23085731ea4a6f